### PR TITLE
WT-8503 Create a function to execute csuite tests in evergreen.yml

### PR DESCRIPTION
**Pull request by etienneptl**
This ticket is part of WT-8498, the initial PR is #7277, this PR aims at merging the same changes to the develop branch.
This PR also includes the following tickets that dealt with similar changes:

- WT-8504 (original PR https://github.com/wiredtiger/wiredtiger/pull/7296). Only the changes relevant to csuite testing have been cherry-picked from this PR.
- WT-8512 (original PR: https://github.com/wiredtiger/wiredtiger/pull/7287).
- WT-8519 (original PR: https://github.com/wiredtiger/wiredtiger/pull/7295).
- WT-8520 (original PR: https://github.com/wiredtiger/wiredtiger/pull/7291).

### CODE REVIEW

**Review by Siddhartha8899 — APPROVED** [review]

**Review by tammybailey — APPROVED** [review]
lgtm - nice improvement!

**Review by lukech — APPROVED** [review]
LGTM

File: test/evergreen.yml [hunk 1]
```diff
@@ -1202,501 +1174,341 @@ tasks:
...
-          working_dir: "wiredtiger/cmake_build"
-          script: |
-            set -o errexit
-            set -o verbose
-
-            ${test_env_vars|} $(pwd)/test/csuite/wt3120_filesys/test_wt3120_filesys -b $(pwd)  2>&1
+      - func: "csuite test"
+        vars:
+          test_name: wt3120_filesys
+          test_args: -b $(pwd)
```
Comment (etienneptl) [inline test/evergreen.yml:1270]: nitpick: @fionaochee, can you swap this line with the first to alphabetize?

File: test/evergreen.yml [hunk 2]
```diff
@@ -1202,501 +1174,341 @@ tasks:
...
-          working_dir: "wiredtiger/cmake_build"
-          script: |
-            set -o errexit
-            set -o verbose
-
-            ${test_env_vars|} $(pwd)/test/csuite/wt2909_checkpoint_integrity/test_wt2909_checkpoint_integrity -b $(pwd) 2>&1
+      - func: "csuite test"
+        vars:
+          test_name: wt2909_checkpoint_integrity
+          test_args: -b $(pwd)
```
Comment (etienneptl) [inline test/evergreen.yml:1451]: Nitpick: alphabetize here too

**Review by etienneptl — COMMENTED** [review]
Thank you so much for taking over and doing the merge, excellent work, LGTM!

**Pull request merged**